internal/sqlparser/testdata/valid-up/test06/04.golden.sql

```diff
@@ -0,0 +1,15 @@
+CREATE FUNCTION do_something(sql TEXT) RETURNS INTEGER AS $$
+BEGIN
+  -- initiate technology
+  PERFORM something_or_other(sql);
+
+  -- increase technology
+  PERFORM some_other_thing(sql);
+
+  -- technology was successful
+  RETURN 1;
+END;
+$$ LANGUAGE plpgsql;
+
+-- 3 this comment will be preserved
```

Comment:
In isolation, it looks like we should not be including these comments, but I think it's defensible to say this is a user error and they should avoid putting trailing comments before the final `-- +goose StatementEnd` annotation.

```

[...]
-- 3 this comment will be preserved
  -- 4 this comment will be preserved


-- +goose StatementEnd
```

Reply:
One thought is we could do a second pass over the complex statement and trim any empty lines/comments after the final semicolon. I'll keep this in mind, likely in a follow up PR.

Reply:
Added logic in https://github.com/pressly/goose/pull/446/commits/98ca49954c46a9e414be7958a7050fdbec868d65 that does a final cleanup of the statement. Effectively this:

```
func cleanupStatement(input string) string {
	if n := strings.LastIndex(input, ";"); n > 0 {
		return input[:n+1]
	}
	return input
}
```

So that this input:

```
-- +goose StatementBegin




-- 1 this comment will NOT be preserved
  -- 2 this comment will NOT be preserved


CREATE FUNCTION do_something(sql TEXT) RETURNS INTEGER AS $$
BEGIN
  -- initiate technology
  PERFORM something_or_other(sql);

  -- increase technology
  PERFORM some_other_thing(sql);

  -- technology was successful
  RETURN 1;
END;
$$ LANGUAGE plpgsql;

-- 3 this comment will NOT be preserved
  -- 4 this comment will NOT be preserved


-- +goose StatementEnd
```

Will end up as the following statement:

```
CREATE FUNCTION do_something(sql TEXT) RETURNS INTEGER AS $$
BEGIN
  -- initiate technology
  PERFORM something_or_other(sql);

  -- increase technology
  PERFORM some_other_thing(sql);

  -- technology was successful
  RETURN 1;
END;
$$ LANGUAGE plpgsql;
```